src/app-modules/docker-compose

```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+# Copyright 2023 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+set -x
+
+docker_compose_cmd=docker-compose
+docker_cmd=docker
+
+app_load() {
+    local application_name="$1"
+    local url="$2"
+    local input_file="$3"
+
+    $docker_cmd image load < "$input_file"
+}
+
+app_save() {
+    local application_name="$1"
+    local url="$2"
+    local output_file="$3"
+
+    $docker_cmd image save "$url" -o "$output_file"
+}
+
+app_rollout() {
+    local application_name="$1"
+    local manifests_dir="$2"
+
+    $docker_compose_cmd --project-directory "$manifests_dir" up -d > "$manifests_dir"/compose.log 2>&1 3>&1 &
+}
+
+app_stop() {
+    local application_name="$1"
+    local manifests_dir="$2"
+
+    $docker_compose_cmd --project-directory "$manifests_dir" down >> "$manifests_dir"/compose.log 2>&1
+}
+
+case "$1" in
+    SAVE)
+        shift
+        app_save "$@"
+        ;;
+    LOAD)
+        shift
+        app_load "$@"
+        ;;
+    ROLLOUT)
+        shift
+        app_rollout "$@"
+        ;;
+    STOP)
+        shift
+        app_stop "$@"
+        ;;
+    EXPORT_MANIFEST)
+        true
+        ;;
+    LS_COMPONENTS)
+        true
+        ;;
+    DELETE)
+        true
+        ;;
+    ROLLBACK)
+        true
+        ;;
+    PURGE)
+        true
+        ;;
+    START)
+        true
+        ;;
+    ALIVE)
+        true
+        ;;
+    HEALTHY)
+        true
+        ;;
+    *)
+        echo "ERROR: unknown API call: $@"
```

Comment:
```suggestion
        echo "ERROR: unknown API call: $1"
```

Reply:
This went missing.

Reply:
> This went missing.

pardon me? I am sorry, I do not get it, it is in the code, you want me to do something here?